tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/service/PostService.java

```diff
@@ -173,9 +174,18 @@ Page<PostViewDTO> findByNationalPostNumber(String query, List<ColumnFilter> colu
    * Method that will throw a not authorized exception if the current logged in user cannot view or
    * modify the person record
    *
-   * @param personId the db managed id of the person record
+   * @param postId the db managed id of the post record
    */
-  void canLoggedInUserViewOrAmend(Long personId);
+  void canLoggedInUserViewOrAmend(Long postId);
+
+  /**
+   * Method that will throw a not authorized exception if the current logged in user's
+   * designated body does not match with post owner.
+   *
+   * @param postId the db managed id of the post record
+   * @param userProfile logged in user's profile
+   */
+  void checkTheLoggedInUserDbSameAsPostOwner(Long postId, UserProfile userProfile);
```

Comment:
May be unneeded as per comment on the implementation class.

Reply:
Yeah, I also thought so but when I call it from PostResource using the "postService" reference then it fails. Interface PostService needs this method here so that PostServiceImpl class can provide the implementation detail.